addons/binding/org.openhab.binding.network.test/src/test/java/org/openhab/binding/network/internal/handler/NetworkHandlerTest.java

```diff
@@ -127,7 +128,7 @@ public void pingDeviceInitTests() {
         // Check that we are online
         ArgumentCaptor<ThingStatusInfo> statusInfoCaptor = ArgumentCaptor.forClass(ThingStatusInfo.class);
         verify(callback).statusUpdated(eq(thing), statusInfoCaptor.capture());
-        Assert.assertThat(statusInfoCaptor.getValue().getStatus(), is(equalTo(ThingStatus.ONLINE)));
+        waitForAssert(() -> assertEquals(statusInfoCaptor.getValue().getStatus(), ThingStatus.ONLINE));
```

Comment:
Ehm... Yeah that works. I mean it is not elegant, but it works. I would have liked if any async behaviour is mocked away or encapsulated with a CompletableFuture that can be waited for, but if you don't want to spend time on that, that's alright I guess.

Reply:
I tried that for the handler tests in the OneWire binding and failed. If you can point me to some example code how this could be done, I'll have alook.